src/main/java/core/basesyntax/service/FileReaderServiceImpl.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.service;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FileReaderServiceImpl implements FileReaderService {
+    @Override
+    public List<String> readFile(String filepath) {
+        List<String> lines = new ArrayList<>();
+        try (BufferedReader br = new BufferedReader(new FileReader(filepath))) {
+            String line;
+            while ((line = br.readLine()) != null) {
+                lines.add(line);
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
Using `e.printStackTrace()` for error handling is not recommended in production code as it doesn't provide a user-friendly error message or recovery mechanism. Consider logging the error using a logging framework or throwing a custom exception to handle it more gracefully.